### PR TITLE
Move version information to pureconfig and remove `whisk_version_name`.

### PR DESCRIPTION
**Pull request by cbickel**
## Description

This PR moves the version information to pureconfig. In addition, it removes the deployment variable `whisk_version_name` as it is only used to determine which type of deployment it is.

## Related issue and scope

- [ ] I opened an issue to propose and discuss this change (#????)

## My changes affect the following components

- [ ] API
- [ ] Controller
- [ ] Message Bus (e.g., Kafka)
- [ ] Loadbalancer
- [ ] Invoker
- [ ] Intrinsic actions (e.g., sequences, conductors)
- [ ] Data stores (e.g., CouchDB)
- [ ] Tests
- [X] Deployment
- [ ] CLI
- [ ] General tooling
- [ ] Documentation

## Types of changes

- [ ] Bug fix (generally a non-breaking change which closes an issue).
- [ ] Enhancement or new feature (adds new functionality).
- [ ] Breaking change (a bug fix or enhancement which changes existing behavior).

## Checklist:

- [X] I signed an [Apache CLA](https://github.com/apache/incubator-openwhisk/blob/master/CONTRIBUTING.md).
- [X] I reviewed the [style guides](https://github.com/apache/incubator-openwhisk/wiki/Contributing:-Git-guidelines#code-readiness) and followed the recommendations (Travis CI will check :).
- [ ] I added tests to cover my changes.
- [ ] My changes require further changes to the documentation.
- [ ] I updated the documentation where necessary.

### CODE REVIEW

**Comment by rabbah** [comment]
Should we add the openwhisk git hash as part of the version info?

File: tests/src/test/scala/actionContainers/ActionContainer.scala [hunk 1]
```diff
@@ -88,9 +88,9 @@ object ActionContainer {
   }
 
   private lazy val dockerCmd: String = {
-    val version = WhiskProperties.getProperty("whisk.version.name")
+    val version = WhiskProperties.getProperty("environment.type")
```
Comment (csantanapr) [inline tests/src/test/scala/actionContainers/ActionContainer.scala]: FYI @jonpspri the version.name is being replaced

**Comment by cbickel** [comment]
PG2#3100 is 🔵

**Comment by rabbah** [comment]
@cbickel rebase.

File: ansible/environments/vagrant/group_vars/all [hunk 2]
```diff
@@ -7,6 +6,8 @@ docker_dns: ""
 runtimes_bypass_pull_for_local_images: true
 invoker_use_runc: "{{ ansible_distribution != 'MacOSX' }}"
 
+environment_type: "distributed"
```
Comment (rabbah) [inline ansible/environments/vagrant/group_vars/all]: Distributed?

**Comment by cbickel** [comment]
PG5#455 🔵

**Pull request merged**